dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantVectorStoreRecordMapper.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.SemanticKernel.Memory;
+using Qdrant.Client.Grpc;
+
+namespace Microsoft.SemanticKernel.Connectors.Qdrant;
+
+/// <summary>
+/// Mapper between a Qdrant record and the consumer data model that uses json as an intermediary to allow supporting a wide range of models.
+/// </summary>
+/// <typeparam name="TRecord">The consumer data model to map to or from.</typeparam>
+internal sealed class QdrantVectorStoreRecordMapper<TRecord> : IVectorStoreRecordMapper<TRecord, PointStruct>
```

Comment:
What is the plan for unit testing this code?

Reply:
We need to refactor the connectors unit test project into separate projects before I can make internals visible to them because with the single project, we run into compiler issues due to multiple copies of InternalUtilities being visible to the unit test projects.
I'll do that in main first, and then rebase the memory feature branch onto main, after which I'll add unit tests in the right project for this and the other new connectors.